pkg/controller/tidb_control.go

```diff
@@ -53,6 +54,10 @@ type TiDBControlInterface interface {
 
 // defaultTiDBControl is default implementation of TiDBControlInterface.
 type defaultTiDBControl struct {
+	// Now, all controllers use this single tidbConrol instance,
+	// we should add a mutex to avoid two controllers update the httpClient.Transport at same time.
+	mutex sync.Mutex
+	// TODO use httpClients cache instead, like pkg/pdapi/pdapi.go does.
```

Comment:
the cache mechanism in `pdapi` has an issue that stale clients will not be freed when the TC cluster is gone. If we want to optimize this, we can cache PD/TiDB clients (per TiDB cluster CR) in the tidb cluster controller instead and free on the deletion of the tidb cluster.

Reply:
Cache the whole client per TiDB cluster in tidb cluster controller is a good idea.